cmd/kubeadm/app/cmd/reset.go

```diff
@@ -198,6 +198,7 @@ func newCmdReset(in io.Reader, out io.Writer, resetOptions *resetOptions) *cobra
 	resetRunner.AppendPhase(phases.NewUpdateClusterStatus())
 	resetRunner.AppendPhase(phases.NewRemoveETCDMemberPhase())
 	resetRunner.AppendPhase(phases.NewCleanupNodePhase())
+	resetRunner.AppendPhase(phases.NewRemoveUsersPhase())
```

Comment:
[Note-to-self]: Add a comment here, explaining why we don't check the feature flag on reset when we remove the users and groups because a user could call `kubeadm init --feature-gates="RootlessControlPlane"` and then call `kubeadm reset` without any feature gates. We should still delete the groups in this case.